easybuild/tools/docs.py

```diff
@@ -160,3 +151,154 @@ def avail_easyconfig_params(easyblock, output_format):
         FORMAT_TXT: avail_easyconfig_params_txt,
     }
     return avail_easyconfig_params_functions[output_format](title, grouped_params)
+
+
+def gen_easyblocks_overview_rst(package_name, path_to_examples, common_params={}, doc_functions=[]):
+    """
+    Compose overview of all easyblocks in the given package in rst format
+    """
+    modules = import_available_modules(package_name)
+    docs = []
+    all_blocks = []
+
+    # get all blocks
+    for mod in modules:
+        for name,obj in inspect.getmembers(mod, inspect.isclass):
+            eb_class = getattr(mod, name)
+            # skip imported classes that are not easyblocks
+            if eb_class.__module__.startswith(package_name) and eb_class not in all_blocks:
+                all_blocks.append(eb_class)
+
+    for eb_class in sorted(all_blocks, key=lambda c: c.__name__):
+        docs.append(gen_easyblock_doc_section_rst(eb_class, path_to_examples, common_params, doc_functions, all_blocks))
+
+    title = 'Overview of generic easyblocks'
+
+    heading = [
+        '=' * len(title),
+        title,
+        '=' * len(title),
+        '',
+        '.. contents::',
```

Comment:
the contents block is pretty long due to the number of sections, I think a long list of links to subsections separated by ` - ` would be nicer?

that does imply that you make sure that each section gets a label: `.. _this_is_a_label:`, which you can refer to using:

```
:ref:`this_is_a_label`
```